addons/sourcemod/scripting/szf/classes.sp

```diff
@@ -0,0 +1,289 @@
+enum struct SurvivorClasses
+{
+	TFClassType nClass;
+	bool bEnabled;
+	float flSpeed;
+	int iRegen;
+	int iAmmo;
+}
+
+enum struct ZombieClasses
+{
+	TFClassType nClass;
+	bool bEnabled;
+	float flSpeed;
+	int iRegen;
+	int iDegen;
+	float flSpree;
+	float flHorde;
+	float flMaxSpree;
+	float flMaxHorde;
+	int iIndex;
+	char sAttribs[256];
+}
+
+enum struct InfectedClasses
+{
+	Infected nInfected;
+	TFClassType nClass;
+	bool bEnabled;
+	float flSpeed;
+	int iRegen;
+	int iDegen;
+	int iIndex;
+	char sAttribs[256];
+}
+
+TFClassType[view_as<int>(TFClassType)] g_nSurvivorClass;
+TFClassType[view_as<int>(TFClassType)] g_nZombieClass;
+Infected[view_as<int>(Infected)] g_nInfectedClass;
+
+static SurvivorClasses g_SurvivorClasses[view_as<int>(TFClassType)];
+static ZombieClasses g_ZombieClasses[view_as<int>(TFClassType)];
+static InfectedClasses g_InfectedClasses[view_as<int>(Infected)];
+
+static ArrayList g_aSurvivorClasses;
+static ArrayList g_aZombieClasses;
+static ArrayList g_aInfectedClasses;
+
+void Classes_Setup()
+{
+	g_aSurvivorClasses = Config_LoadSurvivorClasses();
+	
+	int iCurrent;
+	int iLength = g_aSurvivorClasses.Length;
+	for (int i = 0; i < iLength; i++)
+	{
+		SurvivorClasses sur;
+		g_aSurvivorClasses.GetArray(i, sur);
+		
+		if (sur.bEnabled)
+		{
+			g_nSurvivorClass[iCurrent] = sur.nClass;
+			iCurrent++;
+		}
+		
+		g_SurvivorClasses[sur.nClass].bEnabled = sur.bEnabled;
+		g_SurvivorClasses[sur.nClass].flSpeed = sur.flSpeed;
+		g_SurvivorClasses[sur.nClass].iRegen = sur.iRegen;
+		g_SurvivorClasses[sur.nClass].iAmmo = sur.iAmmo;
+	}
+	
+	g_aZombieClasses = Config_LoadZombieClasses();
+	
+	iLength = g_aZombieClasses.Length;
+	iCurrent = 0;
+	for (int i = 0; i < iLength; i++)
+	{
+		ZombieClasses zom;
+		g_aZombieClasses.GetArray(i, zom);
+		
+		if (zom.bEnabled)
+		{
+			g_nZombieClass[iCurrent] = zom.nClass;
+			iCurrent++;
+		}
+		
+		g_ZombieClasses[zom.nClass].bEnabled = zom.bEnabled;
+		g_ZombieClasses[zom.nClass].flSpeed = zom.flSpeed;
+		g_ZombieClasses[zom.nClass].iRegen = zom.iRegen;
+		g_ZombieClasses[zom.nClass].iDegen = zom.iDegen;
+		g_ZombieClasses[zom.nClass].flSpree = zom.flSpree;
+		g_ZombieClasses[zom.nClass].flHorde = zom.flHorde;
+		g_ZombieClasses[zom.nClass].flMaxSpree = zom.flMaxSpree;
+		g_ZombieClasses[zom.nClass].flMaxHorde = zom.flMaxHorde;
+		g_ZombieClasses[zom.nClass].iIndex = zom.iIndex;
+		strcopy(g_ZombieClasses[zom.nClass].sAttribs, 256, zom.sAttribs);
```

Comment:
All of those can be replaced to `g_ZombieClasses[zom.nClass] = zom;`, copies entire struct into another struct.